packages/swapper/src/swappers/ChainflipSwapper/utils/getQuoteOrRate.ts

```diff
@@ -349,6 +356,11 @@ export const getQuoteOrRate = async (
       singleQuoteResponse.egressAmountNative!,
     )
 
+    const buyAmountBeforeFeesCryptoBaseUnit = toBaseUnit(
+      bnOrZero(singleQuoteResponse.ingressAmount!).times(singleQuoteResponse.estimatedPrice!),
```

Comment:
ditto `ingressAmountNative`

Reply:
Mmmh, estimatedPrice is a precision amount, while ingressAmountNative is a baseUnit amount

I would need to convert estimatedPrice to baseUnit so, it would remain the same don't you think?

Reply:
Not sure what we mean here re: precision amount? `estimatedPrice` is a fiat value, not a crypto value, there is no precision to deal with (though interestingly with weird 6dp): 
<img width="260" alt="image" src="https://github.com/user-attachments/assets/7ee450a9-dd3a-4f49-b2c2-3d0b879de774" />

What I'm suggesting here is instead of converting to base unit, we use a base unit amount directly. There should be no change in the implementation other than that:

```typescript
      const buyAmountBeforeFeesCryptoBaseUnit = bnOrZero(
        singleQuoteResponse.boostQuote.ingressAmountNative!,
      )
        .times(singleQuoteResponse.estimatedPrice!)
        .toFixed(0)
```

Reply:
Actually disregard me, `ingressAmountNative` is in sell asset base unit while we're trying to get an amount in buy asset base unit.